third_party/move/move-vm/runtime/src/loader/function.rs

```diff
@@ -270,6 +511,28 @@ impl Function {
         &self.param_tys
     }
 
+    /// Creates the function type instance for this function. This requires cloning
+    /// the parameter and result types.
+    pub fn create_function_type(&self) -> Type {
+        Type::Function {
+            args: self.param_tys.clone(),
+            results: self.return_tys.clone(),
+            abilities: self.abilities(),
+        }
+    }
+
+    /// Returns the abilities associated with this function, without consideration of any captured
+    /// closure arguments. By default, this is copy and drop, and if the function is
+    /// immutable (public), also store.
+    pub fn abilities(&self) -> AbilitySet {
+        let result = AbilitySet::singleton(Ability::Copy).add(Ability::Drop);
+        if !self.is_friend_or_private {
```

Comment:
Do you need to check the captured types as well?

Reply:
Yes, function only. This is intersected with captured abilities when computed for closures.